### PR TITLE
Update able-player with optional vtt caption file

### PR DESCRIPTION
**Pull request by P-Gill97**
https://openstax.atlassian.net/browse/CORE-242

### CODE REVIEW

File: projects/able-player/template/index.html [hunk 1]
```diff
@@ -41,6 +41,10 @@
 
   <body>
     <video data-able-player data-skin="2020" preload="auto" playsinline data-youtube-nocookie data-allow-fullscreen data-youtube-id="<%= ancillary.youtube_url %>">
+    <% if (format.vtt_file) { %>
```
Comment (TomWoodward) [inline projects/able-player/template/index.html]: i want this to support all the different types of captions. they are listed here https://developer.mozilla.org/en-US/docs/Web/API/VideoTrack/kind

we will need to invent a different field name for each one

**Review by TomWoodward — CHANGES_REQUESTED** [review]

**Comment by P-Gill97** [comment]
I added a different field name for each kind of media alternative file. 

I uploaded a zip file with the captions, descriptions, subtitles, metadata and chapter `.vtt` files I have been using for testing. In case you want to use these to test.

[media_files.zip](https://github.com/user-attachments/files/17837681/media_files.zip)

File: projects/able-player/template/index.html [hunk 2]
```diff
@@ -40,7 +40,28 @@
...
+        <track kind="chapters" src="<%= fieldUrl('chapters') %>" srclang="en" label="chapters" />
+        </track>
+      <% } %>
+      <% if (format.descriptions) { %>
+        <track kind="descriptions" src="<%= fieldUrl('descriptions') %>" srclang="en" label="descriptions" />
+        </track>
+      <% } %>
+      <% if (format.subtitles) { %>
+        <track kind="subtitles" src="<%= fieldUrl('subtitles') %>" srclang="en" label="subtitles" />
+        </track>
```
Comment (TomWoodward) [inline projects/able-player/template/index.html]: these track elements are all malformed, they have self closing opening tags and closing tags. examples on mdn and able-player docs both use self-closing track tags, so i think that is probably the way to do it. can remove the closing tags

also apparently "Subtitles" applies to only translations and that is why they require a language attribute. we don't have the ability to set that up in the admin right now (multiple subtitles with different associated languages) so lets skip subtitles for now

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: projects/able-player/template/index.html (hunk 2 above)
Comment (P-Gill97) [reply projects/able-player/template/index.html]: I removed the redundant closing tags and removed srclang and the subtitles track.

**Review by TomWoodward — APPROVED** [review]

**Comment by TomWoodward** [comment]
please upload this to ram dev and make the updates on ancillary-service dev's video type to make this work

**Pull request merged**